core/src/main/java/org/apache/accumulo/core/util/Merge.java

```diff
@@ -256,7 +257,7 @@ protected Iterator<Size> getSizeIterator(AccumuloClient client, String tablename
       return tablets.stream().map(tm -> {
         long size = tm.getFilesMap().values().stream().mapToLong(DataFileValue::getSize).sum();
         return new Size(tm.getExtent(), size);
-      }).iterator();
+      }).collect(Collectors.toList()).iterator();
```

Comment:
This getSizeIterator method is only used in one place.  Could possibly inline it and create the TabletMetadata in a try w/ resource in that method.  Thinking that could avoid buffering the data into a list.

Reply:
I did this in e4c3dad but it broke MergeTest because the method removed, `getSizeIterator`, was being overridden by the test. Looking into a fix for that now but your suggestion should be implemented.